### PR TITLE
Code quality tools

### PR DESCRIPTION
**Pull request by jodosha**
This PR:

  - [x] Updates broken `Gem version` badge
  - [x] Updates broken `Build status` badge in favor of `TravisCI`
  - [x] Adds a `CircleCI` badge
  - [x] Enable CodeCov for this repository (with a badge)
  - [x] Enables depfu for this repository (with a badge)
  - [x] Removes CodeClimate
  - [x] Removes Gemnasium
  - [x] Removes Coveralls
  - [x] Bumps rubies for TravisCI
  - [x] Setups CircleCI

### CODE REVIEW

File: script/ci [hunk 1]
```diff
@@ -2,8 +2,24 @@
...
+  fi
+}
+
+print_ruby_version() {
+  echo "Using $(ruby -v)"
+  echo
+}
+
+run_code_quality_checks() {
+  # bundle exec rubocop .
```
Comment (AlfonsoUceda) [inline script/ci]: is this ok?

File: .travis.yml [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
+script: ./script/ci
 rvm:
-  - 2.3.6
-  - 2.4.2
-  - 2.5.0
-  - jruby-9.1.13.0
+  - 2.4.4
+  - 2.3.7
+  - 2.5.1
+  - jruby-9.1.9.0
```
Comment (AlfonsoUceda) [inline .travis.yml]: jruby-9.2.0.0?

File: script/ci (hunk 1 above)
Comment (GustavoCaso) [reply script/ci]: From what I can see from other projects like `router` or `controller` they have the same code.
Maybe the code quality is run with another tool?

Comment (jodosha) [reply script/ci]: We don't lint all the Hanami gems with `rubocop` yet, as their 1.x version. At 2.x will be all linted.

**Pull request merged**